src/PrivateComposerInstaller/Env.php

```diff
@@ -10,35 +10,67 @@
 
 class Env
 {
-    protected $arrayAdapter;
+    /**
+     * @var ArrayAdapter
+     */
+    protected $dotenvAdapter;
+
+    /**
+     * @var PutenvAdapter
+     */
     protected $getenvAdapter;
-    protected $path;
-    protected $loaded = false;
 
-    public function __construct($path)
+    /**
+     * @var string
+     */
+    protected $dotenvPath;
+
+    /**
+     * Constructor
+     * @param string $dotenvPath Path to the .env file
+     */
+    public function __construct(string $dotenvPath)
     {
-        $this->arrayAdapter = new ArrayAdapter();
         $this->getenvAdapter = new PutenvAdapter();
-        $this->path = $path;
+        $this->dotenvPath = $dotenvPath;
     }
 
-    public function load()
+    /**
+     * Lazily creates an ArrayAdapter instance providing .env file variables.
+     * @return ArrayAdapter
+     */
+    protected function getDotenvAdapter(): ArrayAdapter
     {
-        $this->loaded = true;
-        $loader = new Loader([$this->path], new DotenvFactory([$this->arrayAdapter]));
-        $loader->load();
-    }
+        if ($this->dotenvAdapter === null) {
+            $this->dotenvAdapter = new ArrayAdapter();
 
-    public function get($key)
-    {
-        $this->getenvAdapter->get($key)->getOrCall(function() use ($key) {
-            if (!$this->loaded) {
-                $this->load();
+            // Load the .env file if it exists or leave the array adapter empty
+            if (file_exists($this->dotenvPath)) {
```

Comment:
Better to catch the path exception from the library than to do this check yourself, since reading could fail for other reasons, or there could be a race condition where the file is deleted.

Reply:
Indeed, I'll fix that. Thank you!